core/trino-main/src/test/java/io/trino/operator/scalar/TestJsonStringArrayExtractScalar.java

```diff
@@ -0,0 +1,187 @@
+import org.junit.jupiter.api.parallel.Execution;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+import static io.trino.operator.scalar.ArrayTransformFunction.ARRAY_TRANSFORM_NAME;
+import static io.trino.operator.scalar.JsonStringArrayExtractScalar.JSON_STRING_ARRAY_EXTRACT_SCALAR_NAME;
+import static io.trino.spi.StandardErrorCode.INVALID_CAST_ARGUMENT;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static io.trino.sql.analyzer.TypeSignatureProvider.fromTypes;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.any;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.anyTree;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.output;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.project;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.values;
+import static io.trino.testing.assertions.TrinoExceptionAssert.assertTrinoExceptionThrownBy;
+import static io.trino.type.JsonType.JSON;
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+import static org.junit.jupiter.api.parallel.ExecutionMode.CONCURRENT;
+
+@TestInstance(PER_CLASS)
+@Execution(CONCURRENT)
+class TestJsonStringArrayExtractScalar
+{
+    private static final TestingFunctionResolution FUNCTIONS = new TestingFunctionResolution();
+    private static final ResolvedFunction TRANSFORM = FUNCTIONS.resolveFunction(ARRAY_TRANSFORM_NAME, fromTypes(new ArrayType(JSON), new FunctionType(List.of(JSON), VARCHAR)));
+    private static final ResolvedFunction JSON_STRING_ARRAY_EXTRACT_SCALAR = FUNCTIONS.getMetadata().resolveBuiltinFunction(JSON_STRING_ARRAY_EXTRACT_SCALAR_NAME, fromTypes(VARCHAR, JsonPathType.JSON_PATH));
+    private QueryAssertions assertions;
+
+    @BeforeAll
+    public void init()
+    {
+        assertions = new QueryAssertions();
+    }
+
+    @AfterAll
+    public void teardown()
+    {
+        assertions.close();
+        assertions = null;
+    }
+
+    @Test
+    void testEmptySliceExtract()
+    {
+        assertTrinoExceptionThrownBy(() -> assertions.expression("transform(cast(json_parse(a) as array<json>), x -> json_extract_scalar(\"x\", '$.name'))")
+                .binding("a", "''").evaluate())
+                .hasErrorCode(INVALID_CAST_ARGUMENT)
+                .hasMessageContaining("Cannot cast to array(varchar). Expected a json array, but got null");
+    }
+
+    @Test
+    void testEmptyStringExtract()
+    {
+        failedExtract("", "$");
+    }
+
+    @Test
+    void testObjectStringExtract()
+    {
+        failedExtract("{}", "$");
+    }
+
+    @Test
+    void testScalarStringExtract()
+    {
+        failedExtract("1", "$");
+    }
+
+    @Test
+    void testEmptyArrayExtract()
+    {
+        Result result = extract("[]", "$.name");
+        assertThat(result.value()).isEqualTo(List.of());
```

Comment:
`isEqualTo(List.of());` -> `isEmpty()`

Reply:
`result.value` is `Object` and it will require to cast